libs/vkd3d/command.c

```diff
@@ -9473,21 +9473,55 @@ static void STDMETHODCALLTYPE d3d12_command_list_ClearUnorderedAccessViewFloat(d
     d3d12_command_list_clear_uav(list, &d, resource_impl, &args, &color, rect_count, rects);
 }
 
+static bool d3d12_command_list_is_subresource_bound_as_rtv_dsv(struct d3d12_command_list *list,
+        struct d3d12_resource *resource, const VkImageSubresource *subresource)
+{
+    unsigned int i;
+
+    if (!(list->rendering_info.state_flags & (VKD3D_RENDERING_ACTIVE | VKD3D_RENDERING_SUSPENDED)))
+        return false;
+
+    if (list->dsv.resource == resource)
```

Comment:
I assume this is okay even if the RTV mask / DSV mask does not bind every render target? I believe we perform the begin/end barriers either way regardless?

Reply:
Yes, this is why this behaves differently than the clear stuff.

Reply:
OK.